pkg/k8s/node.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"fmt"
+	"net"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// GetNodeAddr gets the available IP address of a Node. GetNodeAddr will first try to get the NodeInternalIP, then try
+// to get the NodeExternalIP.
+// Note: Although K8s supports dual-stack, there is only a single Internal address per Node because of issue (
+// kubernetes/kubernetes#91940 ). The Node might have multiple addresses after the issue is fixed, and one per address
```

Comment:
I notice that this issue (https://github.com/kubernetes/kubernetes/issues/91940) has been fixed upstream, so maybe we should open an issue to address this and update `GetNodeAddr`

Reply:
While I was about to open an issue, I found I was not sure what change we need to make here. I see current usages of this function expects a Node IP so they can use it to create tunnel, identify network interface, or connect to the server. So it seems fine if there are two internal addresses or two external addresses, except the returned address family is not determinate. What’s the required change on your mind?

Reply:
I may be wrong, but don't we need to get both an IPv4 address and an IPv6 address for the Node if we want to support dual-stack in noEncap mode? So this function should be able to return both addresses?

Right now we use the one Node IP to install both routes in noEncap mode, but that seems wrong to me. I don't think we have tested noEncap for dual-stack clusters and my expectation is that it may not work:

https://github.com/antrea-io/antrea/blob/fa5419076ef6eac62e7669e76233ff4df1bbe814/pkg/agent/controller/noderoute/node_route_controller.go#L485-L490

Adding @lzhecheng as this may be related to https://github.com/antrea-io/antrea/issues/136#issuecomment-844269436

Reply:
I see, you mean we need to setup one IPv4 route and one IPv6 route for dual-stack clusters. Since https://github.com/antrea-io/antrea/issues/136 is still open, maybe we could just use that issue to track.

Reply:
That works for me. Thanks for updating the issue.